Update README (#8333)
LiskHQ/lisk-sdk: sdk/README.md

[![codecov](https://codecov.io/gh/LiskHQ/lisk-sdk/branch/development/graph/badge.svg?token=2JhT7caf5x)](https://codecov.io/gh/LiskHQ/lisk-sdk)
[![License: Apache 2.0](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)

## Sapphire phase nearly completed

With the release of Lisk SDK v6.0.0, we are completing the [Sapphire phase](https://lisk.com/roadmap), which brings the [Lisk interoperability solution](https://lisk.com/blog/research/lisk-interoperability-solution-published).

> To learn more about the Lisk protocol and its' current development status, check out [Lisk Roadmap](https://lisk.com/roadmap), [Lisk Improvement Proposal](https://github.com/LiskHQ/lips), and [Research Forum](https://research.lisk.com/).

Reviewer: ```suggestion
To learn more about the Lisk protocol and its current development status, check out [Lisk Roadmap](https://lisk.com/roadmap), [Lisk Improvement Proposal](https://github.com/LiskHQ/lips), and [Research Forum](https://research.lisk.com/).
```

> We hope you enjoy building your proof-of-concept Web3 applications using the Lisk SDK, and look forward to receive your feedback and contributions during the beta phase.

Reviewer: ```suggestion
We hope you enjoy building Web3 applications using the Lisk SDK, and look forward to receiving your feedback and contributions during the beta phase.
```

## What is the Lisk SDK?

> The Lisk SDK aims to provide an easy and reliable software development kit for building blockchain applications which are compatible with the [Lisk Protocol](https://lisk.com/documentation/understand-blockchain/lisk-protocol).

Reviewer: Points to v5 docs. Might be better to link this to the Lips Homepage on github.

Author: removed the link

##### Ubuntu

```sh
sudo apt update
sudo apt install -y libtool automake autoconf curl build-essential python2-minimal
```

##### macOS

Ensure that [Homebrew](https://brew.sh/) is installed.

```sh
brew install autoconf automake libtool python2
```

### Initializing a new blockchain client

Execute the `lisk init` command of Lisk Commander as shown below, to bootstrap a default blockchain client in the current folder.

```sh
lisk init
```

As a result of executing the `init` command, you will be asked for the _Name_, _ChainID_, _Description_, _Author_, and _License_ of the new blockchain client.

```
Using template "lisk-ts"
Initializing git repository
Updating .liskrc.json file
Creating project structure
? Application name my_client
? Chain ID in hex representation. ChainID must be 4 bytes (8 characters) 12345678
? Application description A simple blockchain client for my Web3 application
? Author XYZ
? License ISC
```

Next, all the required files are created by Lisk Commander.

It is now already possible to start the newly bootstrapped blockchain client with default configurations for a local Devnet:

```sh
./bin/run start
```

### Configure your blockchain parameters

> The blockchain client offers a variety of configuration options such as `blockTime`, `maxTransactionsSize` and more.

Reviewer: ```suggestion
The blockchain client offers a variety of configuration options such as `blockTime`, `maxTransactionsSize`, and much more.
```

> For a complete overview of the client configuration options, please check out the [SDK configuration reference](https://lisk.com/documentation/lisk-sdk/v6/config.html) and the [Blockchain client configuration](https://liskhq.github.io/lisk-docs/beta/build-blockchain/configuration.html) guide.

Reviewer: `Blockchain client configuration` points to staging docs. Shouldn't this point to the main site?

To use the custom config with the client, use the `--config` flag.
Also add the `--overwrite-config` flag, if you used another config the last time the client was started.

```sh
./bin/run start --config config/custom_config.json --overwrite-config
```

### Register a custom module or a custom plugin

> A custom [module](https://lisk.com/documentation/beta/understand-blockchain/sdk/modules-commands.html) is a logic to define state changes that will be executed on-chain meaning that it will be a part of the blockchain protocol.
> 
> A custom [plugin](https://lisk.com/documentation/beta/understand-blockchain/sdk/plugins.html), on the other hand, is a logic to define an off-chain logic which is not part of the blockchain protocol but to enhance the blockchain application features.

Reviewer: ```suggestion
A custom [module](https://lisk.com/documentation/beta/understand-blockchain/sdk/modules-commands.html) defines the logic that makes state changes on-chain, meaning that it will be a part of the blockchain protocol.

A custom [plugin](https://lisk.com/documentation/beta/understand-blockchain/sdk/plugins.html), on the other hand, defines an off-chain logic that is not part of the blockchain protocol but enhances the blockchain application features.
```

Add your custom module and custom plugin to your blockchain client conveniently with Lisk Commander:

#### Generating a new module

Use the `generate:module` command in the root folder of the blockchain client to generate a ready-to-use module template.

```sh
lisk generate:module my-module-name
```

For more detailed explanations how to create a module, please refer to the guide [How to create a module](https://lisk.com/documentation/beta/build-blockchain/module/index.html)

#### Generating a new plugin

Use the `generate:plugin` command in the root folder of the blockchain client to generate a ready-to-use plugin template.

```sh
lisk generate:plugin my-plugin-name
```

> For more detailed explanations how to create a module, please refer to the guide [How to create a plugin](https://lisk.com/documentation/beta/build-blockchain/plugin/index.html)

Reviewer: ```suggestion
For a more detailed explanation of how to create a plugin, please refer to the guide: [How to create a plugin](https://lisk.com/documentation/beta/build-blockchain/plugin/index.html)
```

### Documentation

> For further explanations, guides and tutorials, see the [official Lisk SDK documentation](https://lisk.com/documentation/lisk-sdk).

Reviewer: I think the URL should point to `beta` version.

The Lisk SDK makes every effort to allow developers to focus simply and purely on writing the code that matters to their own blockchain application, and nothing more.

## Usage

Lisk SDK is an all-in-one package that provides you with tools to create, run and maintain blockchain applications in JavaScript/TypeScript.

### Dependencies

The following dependencies need to be installed in order to run blockchain clients created with the Lisk SDK.

#### Node.js

If you are using NVM, install the correct version as shown below:

```sh
nvm install v16.18.0
```

#### Lisk Commander

It is recommended to install Lisk Commander globally with NPM (Node Package Manager), to facilitate the convenient usage of the Lisk Commander CLI.

```sh
npm install --global lisk-commander@beta
```

To check the successful installation of Lisk Commander, run the following command:

```sh
$ lisk --version
lisk-commander/6.0.0 darwin-arm64 node-v16.18.0
```

#### Toolchain dependencies

Once Lisk Commander is installed, install the following compiler dependencies as well.

##### Ubuntu

```sh
sudo apt update
sudo apt install -y libtool automake autoconf curl build-essential python2-minimal
```

> ##### macOS

Reviewer 2: i think this section is only needed for M1~ ?

Reviewer 2: in the non M1 mac, i think we didn't need this setup 🤔

Reviewer 2: but i think having them will not harm anything

? Application name my_client
? Chain ID in hex representation. ChainID must be 4 bytes (8 characters) 12345678
? Application description A simple blockchain client for my Web3 application
? Author XYZ
? License ISC
```

Next, all the required files are created by Lisk Commander.

It is now already possible to start the newly bootstrapped blockchain client with default configurations for a local Devnet:

```sh
./bin/run start
```

### Configure your blockchain parameters

The blockchain client offers a variety of configuration options such as `blockTime`, `maxTransactionsSize`, and much more.

For a complete overview of the client configuration options, please check out the [SDK configuration reference](https://lisk.com/documentation/lisk-sdk/v6/config.html) and the [Blockchain client configuration](https://lisk.com/documentation/beta/build-blockchain/configuration.html) guide.

To use the custom config with the client, use the `--config` flag.
Also add the `--overwrite-config` flag, if you used another config the last time the client was started.

```sh
./bin/run start --config config/custom_config.json --overwrite-config
```

### Register a custom module or a custom plugin

> A custom [module](https://lisk.com/documentation/beta/understand-blockchain/sdk/modules-commands.html) defines the logic that makes state changes on-chain, meaning that it will be a part of the blockchain protocol.

Reviewer 2: I think we don't use `custom` anymore. we can simply call `a new module` or `a new plugin`

> 

Reviewer: **LINE 209**

**Note:** In case you face any issues during the installation, make sure you have the right version of `yarn` and `node` and try resetting the project with `yarn clean:full`.
